lib/mix/lib/mix/dep/lock.ex

```diff
@@ -27,7 +27,7 @@ defmodule Mix.Dep.Lock do
   @doc """
   Receives a map and writes it as the latest lock.
   """
-  @spec write(map) :: :ok
+  @spec write(map | keyword) :: :ok
```

Comment:
This could be fixed at the call site by converting to a map before calling `Lock.write`

Reply:
fixed